packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts

```diff
@@ -521,17 +536,22 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
             const { summaryVersion } = await summarize(summarizer);
             const dataObject = await getTombstonedDataObjectFromSummary(summaryVersion, unreferencedId);
 
-            // Note: if a user makes a request that looks like this, we will also think that the request is via handle
-            const request: IRequest = { url: unreferencedId, headers: { [RuntimeHeaders.viaHandle]: true } };
-            const response = await dataObject._context.IFluidHandleContext.resolveHandle(request);
+            const serializer = new FluidSerializer(dataObject._context.IFluidHandleContext, () => {});
+            const handle: IFluidHandle = parseHandles({ type: "__fluid_handle__", url: unreferencedId }, serializer);
 
-            assert(response !== undefined, `Expecting a response!`);
-            assert(response.status === 404);
-            assert(response.value === `Datastore removed by gc: ${unreferencedId}`);
+            let tombstoneError: Error & { code: number; underlyingResponseHeaders?: { [TombstoneResponseHeaderKey]: boolean; }; } | undefined;
+            try {
+                await handle.get();
+            } catch (error: any) {
+                tombstoneError = error
+            }
+            assert.equal(tombstoneError?.code, 404, "Tombstone error from handle.get should have 404 status code");
+            assert.equal(tombstoneError?.message, `Datastore removed by gc: ${unreferencedId}`, "Incorrect message for Tombstone error from handle.get");
+            assert.equal(tombstoneError?.underlyingResponseHeaders?.[TombstoneResponseHeaderKey], true, "Tombstone error from handle.get should include the tombstone flag");
```

Comment:
Is there any value in adding retrieving this object via resolveHandle after this to mimic what the actual usage would be?

Reply:
Ok sure